### PR TITLE
Upgrade to wgpu 0.13

### PR DESCRIPTION
**Pull request by JCapucho**
Upstream `wgpu` introduced some changes to how buffers are mapped, so that now instead of returning a future they expect a callback.

This PR accommodates those changes so that the crate is ready for when 0.13 lands and for easier discoverability for those who want to use wgpu master.

### CODE REVIEW

**Comment by Wumpf** [comment]
Thank you so much for contributing and even putting up the draft PR before those wgpu changes even landed! 🤩 
I'll have a closer look once those landed and we're ready to publish (:

**Comment by Imberflur** [comment]
I noticed wgpu added a new feature that must be enabled for using timestamps inside renderpasses. <https://docs.rs/wgpu/latest/wgpu/struct.Features.html#associatedconstant.WRITE_TIMESTAMP_INSIDE_PASSES>

So this line is no longer completely accurate https://github.com/Wumpf/wgpu-profiler/blob/216ce6ab0881d947448d7705900f2a55fb867e54/src/lib.rs#L42

A few options to address this come to my mind:
* Leave it up to the user to manage where the scopes are placed and what features are required.
* Add the new feature flag to this constant ignoring the case where it may not be available.
* Wire up and expose a toggle for emitting timestamp commands inside of passes similar to the current global toggle. It might be neat to have some way to pass the currently enabled features to `GpuProfiler` and have it automatically configure what is enabled based on that.

**Comment by cwfitzgerald** [comment]
I would love the 3rd option - that option won't be available on metal on Apple's M1s, so and this is definitely an important target.

**Comment by JCapucho** [comment]
I've updated this PR to point to the crates.io release version, added the missing feature to `REQUIRED_WGPU_FEATURES` and modified the `new` method to automatically configure the profiler behavior as suggested.

File: src/lib.rs
Comment (Wumpf) [inline src/lib.rs:182]: SeqCst is a bit too much here. I'm quite sure Ordering::Release is the correct ordering semantic here: We want to ensure that anyone seeing this variable updated (i.e. potentially hitting `frame.query_pools.len()`) will see any other writes (i.e. the buffer being mapped)

pondering actually if relaxed might be enough since pending_frames can't be access in parallel while we're in here, meaning frame.query_pools.len() needs no syncing and the semantics of map should be enough to ensure there is a barrier..... but this gives me a headache so let's not go there ;)

File: src/lib.rs
Comment (Wumpf) [inline src/lib.rs:201]: should be Ordering::Acquire, rationale see comment above - here we want to acquire the state that got published earlier

File: src/lib.rs
Comment (Wumpf) [inline src/lib.rs:38]: we probably shouldn't call it "Required" anymore 🤔

**Review by Wumpf — APPROVED** [review]
Excellent! Agreeing with everyone else here that the "toggles solution" as implemented now is best

Left two comments nerding about atomic ordering, but those don't _really_ matter here. So gonna merge, do that atomics myself (to take the `git blame` on me ;)) and publish a new version

File: src/lib.rs
Comment (Wumpf) [reply src/lib.rs:38]: hope @Imberflur isn't too sad if we leave name and doc for it for now :)

Comment (Wumpf) [reply src/lib.rs:38]: ended up "fixing" it in a followup pr

**Pull request merged**
